### ISSUE TITLE
Problems to access the preview of a markdown file using orca

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#143185]
Issue Type: <b>Bug</b>

CC: @isidorn 
CC: @joanmarie

When editing a markdown file, if I activate the previewer by pressing ctrl+shift+v, I cannot browse the displayed content.

Steps to reproduce:

1. Create an empty folder.
1. Inside the folder created in the previous step, create 3 empty files called a.js, b.js and c.js.
I don't know if this is important in reproducing the problem, but this is how I reproduce it.
1. Copy the README.md file present in the VSCode repository to the folder created in the first step.
It may be a false impression, but it seems that the problem is more evident when I use this file.
1. Launch orca.
1. Launch VSCode and open the folder created in the first step.
1. Open the README.md file present in this folder.
1. Activate the preview pressing ctrl+shift+v and  Try to navigate using arrows.

Actual results:
Orca stays mute.

Expected results:
It should be possible to browse the displayed content as if I were browsing a web page.

VS Code version: Code - Insiders 1.65.0-insider (52e65499654de7525e30df3a8228b69320796bf5, 2022-02-16T05:15:13.081Z)
OS version: Linux x64 5.16.9-arch1-1
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz (4 x 2495)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|0, 0, 1|
|Memory (System)|15.54GB (1.48GB free)|
|Process Argv|-n . --crash-reporter-id bcecf5e0-97d1-4830-b28c-0943c4adab1c|
|Screen Reader|yes|
|VM|0%|
|DESKTOP_SESSION|gnome-xorg|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome-xorg|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (37)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|2.2.2
EditorConfig|Edi|0.16.4
prettier-vscode|esb|9.2.0
shell-format|fox|7.2.2
rest-client|hum|0.24.6
vscode-docker|ms-|1.19.0
csharp|ms-|1.24.0
vscode-aks-tools|ms-|0.0.12
vscode-kubernetes-tools|ms-|1.3.6
python|ms-|2022.1.1826356738-dev
vscode-pylance|ms-|2022.2.2-pre.1
remote-containers|ms-|0.222.0
remote-ssh|ms-|0.74.0
remote-ssh-edit|ms-|0.74.0
azure-account|ms-|0.10.0
powershell|ms-|2021.12.0
vsliveshare|ms-|1.0.5330
vsliveshare-audio|ms-|0.1.91
oracledevtools|Ora|21.3.0
vscode-boot-dev-pack|Piv|0.1.0
vscode-concourse|Piv|1.31.0
vscode-manifest-yaml|Piv|1.31.0
vscode-spring-boot|Piv|1.31.0
fabric8-analytics|red|0.3.5
java|red|1.3.0
vscode-commons|red|0.0.6
vscode-xml|red|0.19.1
vscode-yaml|red|1.4.0
sonarlint-vscode|Son|3.3.0
vscodeintellicode|Vis|1.2.17
vscode-java-debug|vsc|0.38.0
vscode-java-dependency|vsc|0.19.0
vscode-java-pack|vsc|0.21.0
vscode-java-test|vsc|0.34.0
vscode-maven|vsc|0.35.0
vscode-spring-boot-dashboard|vsc|0.2.0
vscode-spring-initializr|vsc|0.8.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30422396
pythontb:30258533
pythonvspyt551cf:30291413
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
pythonvspyt200:30323110
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
wslgetstartedc:30421358
vsclayoutctrc:30405799
dsvsc009:30436273
pythonvsnew555:30438690
vsbas813:30436447
vscscmwlcmc:30428972
azactmsalcf:30432849
vscgsvid1:30433757
cppdebug:30437091
dockersurvey3cf:30438169

```

</details>

**Comment by isidorn** [comment]
@jvesouza thanks for filling this! This works fine on macOS. But with VoiceOver there is a command `ctrl+alt+shift+down` to start browsing the preview. Is there something like this that is expected from Orca? To drill down into the webview?
Because I tried to repro with Orca and Orca indeed stays silent, but I am not sure if I am missing something.

Also was this working before? If yes, when did it stop working?

**Comment by jvesouza** [comment]
@isidorn 
Once the preview is activated it should be possible to navigate with the arrows in the displayed content, without any additional intervention.

I can't say it worked before as I started using this feature recently, but I also can't read the information of the extensions when I press the enter key when focusing on the name of one, although I can't say that the two problems are related.

**Comment by joanmarie** [comment]
TL;DR: Did you switch to browse mode prior to pressing the arrow keys?

Details:

I just gave this a try. When the preview first appears, Orca thinks the focus should be on the page tab. I'll look into why in a moment. But given that this is the case, I was on a page tab (not the document) in focus mode. Under those conditions, reading the document indeed does not work. What I did was the following:

1. Switch to Orca's browse mode (Orca + A)
2. Start down arrowing. Orca started presenting the on-screen contents -- including the page tabs, so:
3. Press H to jump to the next heading because the Readme preview had a heading. Orca presented this heading as expected.
4. Continued reading the preview of Readme.md using the arrow keys

I will grant you that this user experience isn't great. Ideally focus would start out in the document and not on the newly-selected/active page tab. I'll look into why this is taking place, and either fix it in Orca or make a request of VSCode. :smile:  But at least for me, the preview of the markdown file can be read by Orca.

**Comment by joanmarie** [comment]
@isidorn: Probably not. Mind you, I'm still investigating what's going on. Lots of accessibility events....

**Comment by joanmarie** [comment]
Every time I try to reproduce the problem, I get different events and see different issues.

* One time the problem was that the document claiming focus has an index in parent of -1 (that indicates the accessible object associated with the document is in a potentially-bad state, such as being destroyed). I could try to work around that in Orca, and it might already be fixed in Chromium, but:
* Some times there is no focus event at all for the document and the focused state is missing. If it's missing from the state set because AT-SPI2 has a stale value, then I can try to work around that in Orca by clearing AT-SPI2's cache. But if the document then continues to claim that it's not focused, Orca shouldn't change what has focus.

@isidorn: While I'm working on workarounds, could you please see if the problem is also present using a fairly recent Chrome/Chromium in Linux rather than Electron?

**Comment by isidorn** [comment]
@joanmarie I just tried with Chrome 98 using https://vscode.dev/github/microsoft/vscode and I can reproduce this. Note that I do not have the latest Orca version. So it does not seem electron related.

**Comment by isidorn** [comment]
Let's reopen to give more time for the investigation on the Orca side

**Comment by jvesouza** [comment]
> TL;DR: Did you switch to browse mode prior to pressing the arrow keys?
No.
> 1. Switch to Orca's browse mode (Orca + A)
> 2. Start down arrowing. Orca started presenting the on-screen contents -- including the page tabs, so:
> 3. Press H to jump to the next heading because the Readme preview had a heading. Orca presented this heading as expected.
> 4. Continued reading the preview of Readme.md using the arrow keys
Confirmed, works as described.
However, if I press ctrl+tab to go back to the readme.md file and then press ctrl+tab to go back to the preview, the previous steps no longer work.
I need to restart orca for the steps described above to work.

**Comment by mjbvz** [comment]
Closing since this issue has not been actively recently and it is also unclear if there is anything to do on the VS Code side
